staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation_test.go

```diff
@@ -489,3 +489,302 @@ func TestCelCompilation(t *testing.T) {
+			expectedCost: 4999997,
+		},
+		{
+			name:         "array of number arrays with all",
+			schema:       arrayOfArraysWithRule("number", "self.all(x, true)"),
+			expectedCost: 3999998,
+		},
+		{
+			name:         "array of string arrays with exists",
+			schema:       arrayOfArraysWithRule("string", "self.exists(x, true)"),
+			expectedCost: 4999997,
+		},
+		{
+			name:         "array of string arrays with all",
+			schema:       arrayOfArraysWithRule("string", "self.all(x, true)"),
+			expectedCost: 3999998,
+		},
+		{
+			name:         "array of boolean arrays with exists",
+			schema:       arrayOfArraysWithRule("boolean", "self.exists(x, true)"),
+			expectedCost: 4999997,
+		},
+		{
+			name:         "array of boolean arrays with all",
+			schema:       arrayOfArraysWithRule("boolean", "self.all(x, true)"),
+			expectedCost: 3999998,
+		},
+		{
+			name:         "array of objects with all",
+			schema:       objectArrayWithRule("self.all(x, true)"),
+			expectedCost: 705882,
+		},
+		{
+			name:         "array of objects with exists",
+			schema:       objectArrayWithRule("self.exists(x, true)"),
+			expectedCost: 882352,
+		},
+		{
+			name:         "map of numbers with exists",
+			schema:       mapWithRule("number", "self.exists(x, true)"),
+			expectedCost: 2999997,
+		},
+		{
+			name:         "map of numbers with all",
+			schema:       mapWithRule("number", "self.all(x, true)"),
+			expectedCost: 2399998,
+		},
+		{
+			name:   "map of numbers with has",
+			schema: mapWithRule("number", "has(self.x)"),
+			// TODO(DangerOnTheRanger): is a cost of 0 to be expected here?
+			expectedCost: 0,
+		},
+		{
+			name:         "map of strings with exists",
+			schema:       mapWithRule("string", "self.exists(x, true)"),
+			expectedCost: 2499997,
+		},
+		{
+			name:         "map of strings with all",
+			schema:       mapWithRule("string", "self.all(x, true)"),
+			expectedCost: 1999998,
+		},
+		{
+			name:         "map of strings with has",
+			schema:       mapWithRule("string", "has(self.x)"),
+			expectedCost: 0,
+		},
+		{
+			name:         "map of booleans with exists",
+			schema:       mapWithRule("boolean", "self.exists(x, true)"),
+			expectedCost: 1874997,
+		},
+		{
+			name:         "map of booleans with all",
+			schema:       mapWithRule("boolean", "self.all(x, true)"),
+			expectedCost: 1499998,
+		},
+		{
+			name:         "map of booleans with has",
+			schema:       mapWithRule("boolean", "has(self.x)"),
+			expectedCost: 0,
+		},
+		{
+			name:         "string with contains",
+			schema:       stringWithRule("self.contains(\"test\")"),
+			expectedCost: 300001,
+		},
+		{
+			name:   "string with startsWith",
+			schema: stringWithRule("self.startsWith(\"test\")"),
+			// TODO(DangerOnTheRanger): is this the expected cost in this scenario?
+			expectedCost: 2,
+		},
+		{
+			name:   "string with endsWith",
+			schema: stringWithRule("self.endsWith(\"test\")"),
```

Comment:
You don't need to escape quotes here. CEL supports single quotes (`'`) and go supports string literals (``` `string with " quote` ```)

Reply:
Whoops, didn't know that, fixed!